yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -522,7 +525,9 @@ private[yarn] class YarnAllocator(
             // This is usually happened when explicitly killing a container, the result will be
             // returned in one AM-RM communication. So query RPC will be later than this completed
             // container process.
-            releasedExecutorLossReasons.put(eid, exitReason)
+            if (!executorsKilledByDriver.contains(eid)) {
```

Comment:
there is a race here where if the executor went down for reason other then us kill after we called kill we won't get the right loss reason.  ie killExecutor just adding it to list to be removed later, if it happened to die for another reason we would miss out on that.   I don't think this is a big deal but we could put in a check here to compare reason.